### ISSUE TITLE
Need your help in generating "X-hmac-signed-body" and "X-hmac-signed-result"

### ISSUE THREAD

**Issue by awaixom** [PSVita-Dev/Reddit-NDK#1]
Hi @PSVita-Dev 

You did a very good job. i am working on creating python wrapper for reddit, by inspecting api endpoints of reddit using android.
i am working on generating "X-hmac-signed-body" and "X-hmac-signed-result" parameters, Can you please help me with generating these 2 parameters. below is the apk source code of reddit and these parameters are generating in file **"/base_source_from_JADX/sources/p013e/p551a/p787h/p788a/p789a/p792j/C8975c.java"** and in function **mo26422c**.

**_app source code_** 
[https://drive.google.com/file/d/1s5xvwd3hR5BQ5zhDl826UR3-JzD25m0c/view?usp=sharing](url)

**_Picture_**
![image](https://user-images.githubusercontent.com/63053679/212046359-260f5235-21bd-4a2a-a271-d4b050e5fc95.png)

Looking forward to hear from you.
thank you so much

**Comment by technicaljicama** [comment]
Hello, first of all sorry that I havent seen this issue sooner.
Basically (for hmac body) the string "Epoch:timestamp|Body:request body" gets hmac sha256 hashed with the key from this repo.
If you got any questions, i will try to answer them asap

**Comment by LesterGreen0117** [comment]
What is User-Agent supposed to be in Hmac Result? I put in the correct one from request header but the hmac hash don't match, is the User-Agent hashed before it is generated?

**Comment by Kangaroo69** [comment]
@PSVita-Dev

**Comment by technicaljicama** [comment]
You hash the whole string, so "Epoch:timestamp|User-Agent:Reddit v..."

**Comment by PartyGhost2** [comment]
Yes, I did that but it's different still. Here's the original hmac that appears in the request header:
X-Hmac-Signed-Result: 1:android:2:1677526958:be40ce003c3c2c2993dc86de44df6294ae93736346d89463b1ee10749312aff9

Using the string "Epoch:1677526958|User-Agent:Reddit/Version2023.07.1/Build790267/Android 12|Client-Vendor-Id:1659fca3-a0a5-41e1-b457-4f29146c8f2d"
It returns with the hmac:
1a335536ac828439770f4ee3141e793e4dc0a9373dbf9f1e1d75ddc62492f65c

The X-Hmac-Signed-Body comes out correct though

**Comment by Kangaroo69** [comment]
Yes, I did that but it's different still. Here's the original hmac that appears in the request header:
X-Hmac-Signed-Result: 1:android:2:1677526958:be40ce003c3c2c2993dc86de44df6294ae93736346d89463b1ee10749312aff9

Using the string "Epoch:1677526958|User-Agent:Reddit/Version2023.07.1/Build790267/Android 12|Client-Vendor-Id:1659fca3-a0a5-41e1-b457-4f29146c8f2d"
It returns with the hmac:
1a335536ac828439770f4ee3141e793e4dc0a9373dbf9f1e1d75ddc62492f65c

The X-Hmac-Signed-Body comes out correct though @PSVita-Dev

**Comment by morris90901** [comment]
> You hash the whole string, so "Epoch:timestamp|User-Agent:Reddit v..."

Hi i'm doing it like you suggested but the value isn't matching for some reason here is an example maybe I'm doing it wrong
![image](https://github.com/technicaljicama/Reddit-NDK/assets/151993437/dee79568-8ea3-4eef-bf1d-47fe506f6331)
